lisa/interrupts.py

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+from __future__ import annotations
+
+import re
+from typing import Dict, List
+
+from lisa import Node
+from lisa.tools import Cat
+
+
+class Interrupt:
+    irq_number: str
+    interrupt_count: List[int]
+    metadata: str
+
+    # 0:         22          0  IR-IO-APIC   2-edge      timer
+    _interrupt_regex = re.compile(
+        r"^\s*(?P<irq_number>\S+):\s+(?P<interrupt_count>[\d+ ]+)\s*(?P<metadata>.*)$"
+    )
+
+    def __init__(
+        self, irq_number: str, interrupt_count: List[int], metadata: str = ""
+    ) -> None:
+        self.irq_number = irq_number
+        self.interrupt_count = interrupt_count
+        self.metadata = metadata
+
+    def __str__(self) -> str:
+        return (
+            f"irq_number : {self.irq_number}, "
+            f"count : {self.interrupt_count}, "
+            f"metadata : {self.metadata}"
+        )
+
+    def __repr__(self) -> str:
+        return self.__str__()
+
+    @staticmethod
+    def get_interrupt_data(node: Node) -> List[Interrupt]:
```

Comment:
Create a `InterruptInspector` tool, and return the `Interrupt` objects. It's confusing to put the collection operations into the single item class.

Reply:
done